web/service/src/main/java/org/mobilitydata/gtfsvalidator/web/service/controller/ValidationController.java

```diff
@@ -93,27 +100,38 @@ public ResponseEntity runValidator(
       }
 
       ValidationJobMetaData jobData = getFeedFileMetaData(message);
-      var jobId = jobData.getJobId();
+      jobId = jobData.getJobId();
+      outputPath = storageHelper.createOutputFolderForJob(jobId);
       var fileName = jobData.getFileName();
 
       var countryCode = storageHelper.getJobMetadata(jobId).getCountryCode();
 
       // copy the file from GCS to a temp directory
       tempFile = storageHelper.downloadFeedFileFromStorage(jobId, fileName);
-      outputPath = storageHelper.createOutputFolderForJob(jobId);
 
       // extracts feed files from zip to temp output directory, validates, and returns
       // the path to the output directory
       validationHandler.validateFeed(tempFile, outputPath, countryCode);
 
+      logger.info("Validation successful. Writing " + executionResultFile + " file");
+      new ExecutionResult("success", "").writeToFile(outputPath);
       // upload the extracted files and the validation results from outputPath to GCS
       storageHelper.uploadFilesToStorage(jobId, outputPath);
 
       return new ResponseEntity(HttpStatus.OK);
     } catch (Exception exc) {
       logger.error("Error", exc);
       Sentry.captureException(exc);
-      throw new ResponseStatusException(HttpStatus.INTERNAL_SERVER_ERROR, "Error", exc);
+      logger.info("Validation failed. Writing " + executionResultFile + " file");
+      new ExecutionResult("error", exc.getMessage()).writeToFile(outputPath);
+      try {
+        // There was a failure but we still try to upload the execution_result file.
+        storageHelper.uploadFilesToStorage(jobId, outputPath);
+      } catch (IOException e) {
+        logger.error("Error: Could not upload to file storage", e);
+      }
```

Comment:
We should look into the scenario where _storageHelper.uploadFilesToStorage_ throws an exception. The uploadFilesToStorage depends on network and Storage reliability; failures should be temporary rather than permanent. This implementation will not upload the file or retry the event's processing. In this case, let the GCP retry as it might be fixed the previous. Otherwise, the consumer will keep waiting for a file that will never appear. To handle this scenario we can move the _storageHelper.uploadFilesToStorage(jobId, outputPath);_ to the finally block

Reply:
It's like nobody throws specific exceptions these days. I will have to discriminate by having separate try-catch for storage and for validation.

Reply:
Sorry, I was also suggesting that all _storageHelper.uploadFilesToStorage_ calls should move to the finally block, but I wasn't explicit enough  :-(

Reply:
I went for a slightly different option. Essentially any error within the validator will return 200 and fill the execution result file. Any error not in the validator (which includes all error related to storage) will return 500 and trigger a retry.